tests/compas/robots/test_model.py

```diff
@@ -69,6 +69,9 @@ def test_programmatic_robot_model():
     urdf = URDF.from_robot(robot)
     robot_reincarnated = RobotModel.from_urdf_string(urdf.to_string())
     assert(['link0', 'joint1', 'link1', 'joint2', 'link2'] == list(robot_reincarnated.iter_chain()))
+    robot.remove_link('link2')
+    robot.remove_joint('joint2')
+    assert ['link0', 'joint1', 'link1'] == list(robot.iter_chain())
```

Comment:
it would be good to add another unit tests that removes a link/joint from the middle of the chain as well.